src/Features/Core/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.cs

```diff
@@ -750,34 +821,59 @@ private DiagnosticData UpdatePosition(DiagnosticData diagnostic, SyntaxTree tree
                 helpLink: diagnostic.HelpLink);
         }
 
-        private static IEnumerable<DiagnosticData> GetDiagnosticData(Document document, TextSpan? span, IEnumerable<Diagnostic> diagnostics)
+        private static IEnumerable<DiagnosticData> GetDiagnosticData(Document document, SyntaxTree tree, TextSpan? span, IEnumerable<Diagnostic> diagnostics)
         {
-            return diagnostics != null ? diagnostics.Where(dx => ShouldIncludeDiagnostic(dx, span)).Select(d => DiagnosticData.Create(document, d)) : null;
+            return diagnostics != null ? diagnostics.Where(dx => ShouldIncludeDiagnostic(dx, tree, span)).Select(d => DiagnosticData.Create(document, d)) : null;
```

Comment:
as you probably saw from change I made below, we don't use cache at all when project diagnostic reports document diagnostic since there is no easy way to support it. we can't do same thing for document diagnostic. so that will be one of difference between v1 and v2.